### PR TITLE
(Closes #2638) add new linkspector workflow

### LINKED ISSUE #2638: Switch link-checker GHA to Linkspector?
It seems that Linkspector (https://github.com/marketplace/actions/run-linkspector-with-reviewdog) is the supported replacement for markdown-link-check (https://github.com/marketplace/actions/markdown-link-check) so we should probably switch to it. That might then fix the problem we've currently got with a false positive on the GOcean-related URL.

### PR DESCRIPTION
**Pull request by arporter**

### CODE REVIEW

**Comment by arporter** [comment]
Addresses the persistent false positive in the link checking by moving to 'linkspector' instead of markdown link checker. They are both running in parallel in this PR but once the reviewer is happy I propose removing 'markdown-link-checker' altogether.
One for @hiker, @LonelyCat124, @TeranIvy or @sergisiso.

File: .github/workflows/linkcheck.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    if: ${{ github.repository != 'stfc/PSyclone-mirror' }}
+    name: runner / linkspector
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run linkspector
+        uses: umbrelladocs/action-linkspector@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          reporter: github-pr-review
```
Comment (sergisiso) [inline .github/workflows/linkcheck.yml:14]: I think I like more the "github-pr-check" reports inlined in the code than the comments (which look quite noisy), but this is subjective. What do you think?

**Review by sergisiso — CHANGES_REQUESTED** [review]
@arporter I agree we should move to the new tool if it's better and the new preferred way. Also I agree in removing the old check (and its config file) in this PR.

File: .github/workflows/linkcheck.yml (hunk 1 above)
Comment (arporter) [reply .github/workflows/linkcheck.yml:14]: Yes, I'm happy to change it.

File: .github/workflows/linkcheck.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: Linkspector
+on: [pull_request]
+jobs:
+  check-links:
+    if: ${{ github.repository != 'stfc/PSyclone-mirror' }}
+    name: runner / linkspector
```
[the remark this answers is not in the document]
Comment (arporter) [reply .github/workflows/linkcheck.yml:6]: Yes, done.

**Comment by arporter** [comment]
All green and lovely now. I remembered that I needed to update the Dev Guide so I've done that now.
Ready for another look.

File: .github/workflows/linkspector.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+dirs:
```
[the remark this answers is not in the document]
Comment (arporter) [reply .github/workflows/linkspector.yml]: All fixed now I think.

**Review by sergisiso — APPROVED** [review]
@arporter All comments have been addressed, the CI issues are gone and the documentation has been updated. This is ready to merge.

**Pull request merged**